star-dash/star-dash/GameEngine/Components/GrappleHookOwnerComponent.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  GrappleHookOwnerComponent.swift
+//  star-dash
+//
+//  Created by Ho Jun Hao on 28/3/24.
+//
+
+import Foundation
+
+class GrappleHookOwnerComponent: Component {
```

Comment:
Why do we need OwnerComponent if grapple hook already has the owner id?

Reply:
sorry, reviewed again and realised there's a grapple hook entity, but would a ownerPlayerId field in the component make more sense?

Reply:
sure. changed